### DISCUSSION TITLE
How to rescale dB range of spectrogram now that decibel_limits is deprecated

### DISCUSSION THREAD

**Discussion by fascimare** [kitzeslab/opensoundscape#919] [Q&A] [answered]
Hi all,
I am relatively new to opensoundscape, I downloaded the repository two weeks ago. In the tutorials, the authors describe how to set decibel limits for the spectrogram in a pipeline using decibel_limits. However, this parameter has recently been removed from the code. How do I now set dB limits for a spectrogram? Currently, the spectrograms coming out of the pipeline are either all white or all black.

**Comment by sammlapp** [comment, marked as answer]
Hi, welcome to the discussions page @fascimare and thanks for your question. I’ll copy my reply from the related issue #918 

in opensoundscape v0.10.0 this adjustment knob moved from to_spec to to_tensor action (because the decibel limits are now applied not when creating the spectrogram but when converting it to a PyTorch tensor). It looks like the preprocessing tutorial notebook was not updated to reflect this change, apologies about that. We’ll make that change soon. 

To adjust the decibel limits with opensoundscape>=0.10.0 use

`preprocessor.pipeline.to_tensor.set(range=(-150,-50))`

note that the param is called range. 

I’m relying from my phone so apologies if there is a typo, but I believe that line of code will work

**Reply by fascimare** [reply]
This worked! Thank you very much.
